src/domain/core/src/services/verification_service.rs

```diff
@@ -55,6 +55,12 @@ pub enum VerificationResult {
     Valid,
 }
 
+#[derive(Debug)]
+pub struct VerificationMultiResult {
+    pub verification_result: Arc<Result<VerificationResult, VerificationError>>,
```

Comment:
I strongly don't think an `Arc` is required here. It's not something we would share between threads.